cartographer/sensor/voxel_filter.cc

```diff
@@ -77,25 +78,32 @@ PointCloud AdaptivelyVoxelFiltered(
 
 }  // namespace
 
-PointCloud VoxelFiltered(const PointCloud& point_cloud, const float size) {
-  VoxelFilter voxel_filter(size);
-  voxel_filter.InsertPointCloud(point_cloud);
-  return voxel_filter.point_cloud();
-}
-
 VoxelFilter::VoxelFilter(const float size) : voxels_(size) {}
 
-void VoxelFilter::InsertPointCloud(const PointCloud& point_cloud) {
-  for (const Eigen::Vector3f& point : point_cloud) {
+PointCloud VoxelFilter::Filter(const PointCloud& point_cloud) {
```

Comment:
template implementation of this? You can also call .head<3> on Vector3 at no cost.